SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskProcessBuilder.java

```diff
@@ -148,6 +151,16 @@ private ProcessBuilder buildProcessBuilder(TaskInfo taskInfo, ExecutorData execu
     return processBuilder;
   }
 
+  private String serviceLogOutPath() {
+    if (configuration.getTaskAppDirectory().equals(".")) {
+      return configuration.getServiceLog();
+    }
+    Path basePath = task.getTaskDefinition().getTaskDirectoryPath();
+    String appPath = configuration.getTaskAppDirectory().startsWith("./") ? configuration.getTaskAppDirectory().replace("./", "") : configuration.getTaskAppDirectory();
+    Path app = basePath.resolve(appPath);
```

Comment:
you should be able to collapse these two lines to just `basePath.resolve(configuration.getTaskAppDirectory()).normalize()`